### PR TITLE
Make sure WGPU buffers are created with initial data if available

### PR DESCRIPTION
**Pull request by aleino-nv**

### CODE REVIEW

File: src/rhi-shared.h [hunk 1]
```diff
@@ -437,7 +437,7 @@ class SimpleShaderObjectData
 
     Index getCount() { return m_ordinaryData.size(); }
     void setCount(Index count) { m_ordinaryData.resize(count); }
-    uint8_t* getBuffer() { return m_ordinaryData.data(); }
+    uint8_t* getBuffer() { return m_ordinaryData.empty() ? nullptr : m_ordinaryData.data(); }
```
Comment (skallweitNV) [inline src/rhi-shared.h:440]: probably not needed as we shouldn't access data if the buffer is empty .. but ok to keep too

File: src/wgpu/wgpu-shader-object.cpp [hunk 2]
```diff
@@ -402,7 +402,8 @@ Result ShaderObjectImpl::_ensureOrdinaryDataBufferCreatedIfNeeded(
         bufferDesc.defaultState = ResourceState::ConstantBuffer;
         bufferDesc.memoryType = MemoryType::DeviceLocal;
         ComPtr<IBuffer> buffer;
-        SLANG_RETURN_ON_FAIL(encoder->m_device->createBuffer(bufferDesc, nullptr, buffer.writeRef()));
+        void * initData = m_data.getBuffer();
+        SLANG_RETURN_ON_FAIL(encoder->m_device->createBuffer(bufferDesc, initData, buffer.writeRef()));
```
Comment (skallweitNV) [inline src/wgpu/wgpu-shader-object.cpp]:
```suggestion
        SLANG_RETURN_ON_FAIL(encoder->m_device->createBuffer(bufferDesc, m_constantBufferData.data(), buffer.writeRef()));
```

Comment (aleino-nv) [reply src/wgpu/wgpu-shader-object.cpp]: Done except I check for empty as well.

Comment (skallweitNV) [reply src/wgpu/wgpu-shader-object.cpp]: that's not necessary as we exit early if there is no constant data .. see at the start of the function
```c++
    if (m_constantBufferSize == 0)
    {
        return SLANG_OK;
    }
```

Comment (aleino-nv) [reply src/wgpu/wgpu-shader-object.cpp]: Done.

**Pull request merged**